lib/src/components/NxCollapsibleItemsSelect/NxCollapsibleMultiSelect/__tests__/NxCollapsibleMultiSelect.test.tsx

```diff
@@ -4,355 +4,559 @@
+        expect(options[3]).not.toBeChecked();
       });
-      const toggleAllOptionWrapper = shallowRender.renderProp('renderToggleAllOption')();
 
-      expect(toggleAllOptionWrapper).toHaveProp('isChecked', false);
-    });
+      it('renders all unchecked options if selectedIds is not provided', function () {
+        const view = quickRender(),
+            options = view.getAllByRole('menuitemcheckbox');
 
-    it('renders checked <NxCheckbox> if all options are selected', function () {
-      const shallowRender = getShallow({
-        selectedIds: new Set(['foo', 'bar', null])
+        expect(options[0]).not.toBeChecked();
+        expect(options[1]).not.toBeChecked();
+        expect(options[2]).not.toBeChecked();
+        expect(options[3]).not.toBeChecked();
       });
-      const toggleAllOptionWrapper = shallowRender.renderProp('renderToggleAllOption')();
 
-      expect(toggleAllOptionWrapper).toHaveProp('isChecked', true);
-    });
+      it('calls component\'s onChange callback when an option is toggled', async function() {
+        const user = userEvent.setup(),
+            onChange = jest.fn(),
+            view = quickRender({ onChange }),
+            options = view.getAllByRole('menuitemcheckbox');
+
+        expect(onChange).not.toHaveBeenCalled();
+
+        await user.click(options[1]);
+
+        expect(onChange).toHaveBeenCalledWith(new Set(['foo']), 'foo');
+
+        await user.click(options[2]);
+
+        expect(onChange).toHaveBeenCalledWith(new Set(['bar']), 'bar');
+
+        await user.click(options[1]);
 
-    it('renders disabled <NxCheckbox> if component is disabled', function () {
-      const shallowRender = getShallow({
-        disabled: true
+        expect(onChange).toHaveBeenCalledWith(new Set(['foo']), 'foo');
       });
-      const toggleAllOptionWrapper = shallowRender.renderProp('renderToggleAllOption')();
 
-      expect(toggleAllOptionWrapper).toHaveProp('disabled', true);
-    });
+      describe('tooltip', function() {
+        it('sets customized text if optionTooltipGenerator is defined', async function() {
+          const user = userEvent.setup(),
+              optionTooltipGenerator = jest.fn().mockReturnValue('customized-tooltip'),
+              view = quickRender({ optionTooltipGenerator }),
+              options = view.getAllByRole('menuitemcheckbox');
+
+          await runTimers();
+          expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
+
+          await user.hover(options[1]);
+          await runTimers();
+
+          const tooltip = screen.getByRole('tooltip');
 
-    describe('onChange callback', function () {
-      it('selects all options when no options are selected', function() {
-        const onChangeSpy = jest.fn();
-        const shallowRender = getShallow({
-          onChange: onChangeSpy,
-          selectedIds: new Set()
+          expect(tooltip).toHaveTextContent('customized-tooltip');
         });
```

Comment:
Should test that the correct arguments are passed into `optionTooltipGenerator`

Reply:
Updated here 10a242ecd58c2723c1121e5cf860349fb5d9bca0